vcd/config_test.go

```diff
@@ -743,7 +744,7 @@ func getVcdVersion(config TestConfig) (string, error) {
 	if vcdClient == nil || err != nil {
 		return "", err
 	}
-	err = ProviderAuthenticate(vcdClient, config.Provider.User, config.Provider.Password, config.Provider.Token, config.Provider.SysOrg, config.Provider.ApiToken)
+	err = ProviderAuthenticate(vcdClient, config.Provider.User, config.Provider.Password, config.Provider.Token, config.Provider.SysOrg, config.Provider.ApiToken, config.Provider.ServiceAccountToken)
```

Comment:
During a second review, I saw this and I thought, why don't we pass `config` to `ProviderAuthenticate` so it has less arguments? As it seems that it can be prone to add more and more (like now) and it's definitely a code smell.

But then I saw that `ProviderAuthenticate` is public, is there any reason for that? As it is in `vcd` package and our provider should not export anything IMHO. It looks like a convenience private function to me.

If we would like to refactor it, maybe we could:
- Instead of passing members of `config.Provider` to `ProviderAuthenticate(vcdClient, c.Provider...)`, make `(c *Config) providerAuthenticate(vcdClient)`, or in other words, make `providerAuthenticate` a method of `Config`.
- As the function is used for testing, maybe we could do a wrapper:
```
func (testConfig *TestConfig) providerAuthenticateForTests(client *govcd.VCDClient) error {
	c := &Config{
		User:            testConfig.Provider.User,
		// ...
	}
	return c.providerAuthenticate(client, c)
}
```

As this is a huge refactoring that affects almost everything, I'm just adding this as a note here, maybe we could just leave a TODO or FIXME or something and tackle this more calmly in the next release.

Thoughts? @lvirbalas @Didainius

Reply:
Yeah, agree. It has outgrown original signature. It is just a question of when we do it :)

What I don't remember is why is it public @dataclouder might remember if we need it together with a few other functions `DataSources` and `Resources`.

In general - this is not a library and is not meant to be used as one so we don't care that much about changing signature.

Reply:
I added a TODO above the function as a reminder. I'll look into it but thought we should tackle it in a different PR to keep this one small and easier to review